src/client/src/ui/spotTile/epics/pricingEpics.ts

```diff
@@ -1,23 +1,25 @@
-import { applicationConnected, applicationDisconnected } from 'rt-actions'
-import { Observable } from 'rxjs'
-import { map, switchMapTo, takeUntil } from 'rxjs/operators'
+import { Action } from 'redux'
+import { ofType } from 'redux-observable'
+import { applicationDisconnected } from 'rt-actions'
+import { map, mergeMap, takeUntil } from 'rxjs/operators'
 import { ApplicationEpic } from 'StoreTypes'
-import { SpotTileActions } from '../actions'
-import { SpotPriceTick } from '../model/spotPriceTick'
+import { SpotTileActions, TILE_ACTION_TYPES } from '../actions'
+import PricingService from './pricingService'
 
-const { priceUpdateAction } = SpotTileActions
-type PriceUpdateAction = ReturnType<typeof priceUpdateAction>
+const { priceUpdateAction, subscribeToSpotTile } = SpotTileActions
+type SubscribeToSpotTileAction = ReturnType<typeof subscribeToSpotTile>
+
+export const pricingServiceEpic: ApplicationEpic = (action$, state$, { loadBalancedServiceStub }) => {
+  const pricingService = new PricingService(loadBalancedServiceStub)
 
-export const pricingServiceEpic = (
-  pricesForCurrenciesInRefData: Observable<SpotPriceTick>
-): ApplicationEpic => action$ => {
   return action$.pipe(
-    applicationConnected,
-    switchMapTo<PriceUpdateAction>(
-      pricesForCurrenciesInRefData.pipe(
-        map(priceUpdateAction),
-        takeUntil(action$.pipe(applicationDisconnected))
-      )
-    )
+    ofType<Action, SubscribeToSpotTileAction>(TILE_ACTION_TYPES.SPOT_TILE_SUBSCRIBE),
+    mergeMap((action: SubscribeToSpotTileAction) =>
+      pricingService.getSpotPriceStream({
+        symbol: action.payload
+      })
+    ),
+    map(priceUpdateAction),
+    takeUntil(action$.pipe(applicationDisconnected))
```

Comment:
You should leave this nested. This will unsubscribe the whole stream!

Test by connected and disconnecting the application.

Reply:
look carefully at the level at which takeUntil will unsubscribe.